quinn-proto/src/connection/streams/recv.rs

```diff
@@ -299,10 +299,9 @@ impl<'a> Chunks<'a> {
 
         let mut should_transmit = false;
         // We issue additional stream ID credit after the application is notified that a previously
-        // open stream has finished or been reset and we've therefore disposed of its state.
-        if matches!(state, ChunksState::Finished | ChunksState::Reset(_))
-            && self.streams.side != self.id.initiator()
-        {
+        // open stream has finished or been reset and we've therefore disposed of its state, as
+        // recorded by `stream_freed` calls in `next`.
+        if self.streams.take_max_streams_dirty(self.id.dir()) {
```

Comment:
Hmm, maybe we should bump the abstraction level on `take_max_streams_dirty()`, in terms of its name and also maybe have it take a `&mut Retransmits` as an argument?

Reply:
Good idea; done in a separate commit.